webpack.config.js

```diff
@@ -1,43 +1,77 @@
+const webpack = require('webpack');
+const path = require('path');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const CleanWebpackPlugin = require('clean-webpack-plugin');
 
-const path = require("path");
-const HtmlWebpackPlugin = require("html-webpack-plugin");
+// Our directory path
+const BUILD_DIR = path.join(__dirname, 'dist');
+// App directory
+const APP_URL = path.join(__dirname, 'src');
+// Public directory
+const PUBLIC_URL = path.resolve(__dirname, 'public');
+
+// JS|JSX test regex
+const jsTest = /\.(js|jsx)$/;
+// CSS test regex
+const cssTest = /\.css$/;
+// Image test regex
+const imgTest = /\.(png|jpg|jpeg|gif|svg)$/i;
+// Files to clean
+const filesToClean = [
+  'dist',
+];
+
+// Our application vendor files
+const VENDOR_LIBS = [
+  'react',
+  'react-dom',
+  'react-router-dom',
+];
+
+/** Webpack Pluguins */
+
+// HTMLWebpackPlugin config
+const HWP = new HtmlWebpackPlugin({
+  template: `${APP_URL}/index.html`,
+  filename: `${BUILD_DIR}/index.html`,
+});
+
+/** nstance of the CleanWebpackPlugin */
+const CLEANPLUGIN = new CleanWebpackPlugin(filesToClean);
+
+/** Webpack exports */
 
 module.exports = {
-  entry: "./src/index.js",
+  entry: {
+    bundle: `${APP_URL}/index.js`,
+    vendor: VENDOR_LIBS,
+  },
   output: {
-    path: path.join(__dirname, "/dist"),
-    filename: "index_bundle.js"
+    path: BUILD_DIR,
   },
   module: {
     rules: [
       {
-        test: /\.js$/,
+        test: jsTest,
         exclude: /node_modules/,
-        use: {
-          loader: "babel-loader"
-        },
+        use: 'babel-loader',
       },
       {
-        test: /\.css$/,
-        use: ["style-loader", "css-loader"]
+        test: cssTest,
+        use: ['style-loader', 'css-loader'],
       },
       {
-        test: /\.(png|jpg|jpeg|gif)$/,
-        use: [
-          {
-            loader: 'file-loader',
-            options: {},
-          },
-        ],
+        test: imgTest,
+        use: ['file-loader'],
       },
-    ]
+    ],
+  },
+  plugins: [
+    HWP,
+    CLEANPLUGIN,
+  ],
+  devServer: {
+    compress: true,
+    contentBase: BUILD_DIR,
   },
```

Comment:
Hi @JohnMadakin I think we would also need `historyApiFallback: true` here so that all our route will always be redirected to the index.html file

Reply:
thanks